### PR TITLE
Inconsistencies in Consuming REST guide

### PR DESCRIPTION
**Pull request by KyleAure**
The following inconsistencies have been addressed:

- Should work with whole classes instead of code snippets
- Instruction asking users to create new classes is inconsistent with the other guides (and within the guide itself too)?
- Some code blocks are copyable when they shouldn't or don't need to be copyable
- Do we really need to link to the JSON Binding 1.0 User Guide twice? And, do we need to mention the version?
- Command blocks should be copyable and be made consistent with the other guides
- Revise the names of the JUnit tests to be consistent and so we can easily tell which ones are for JSON-B and which is for JSON-P
- Use mvn compile throughout: Kubik42 mentioned that we should do this. Noting it and Dmitry can elaborate on what it means.
- Mention under 'Starting the service' that the guide is setup to use loose application/config so users will have the service started from /start and making updates to code directly where the changes will be picked by the server dynamically.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -163,76 +162,55 @@ http://json-b.net/users-guide.html[JSON Binding 1.0 Users Guide].
...
-
-`src/main/java/io/openliberty/guides/consumingrest/Consumer.java`:
+The `Artist` and `Album` POJOs are ready for deserialization. To consume the JSON response
+from your REST service create the `Consumer` class in the
+`src/main/java/io/openliberty/guides/consumingrest/Consumer.java` file.
 
 [source, java, indent=0]
 ----
-include::finish/src/main/java/io/openliberty/guides/consumingrest/Consumer.java[tags=!comment;!class]
+include::finish/src/main/java/io/openliberty/guides/consumingrest/Consumer.java[tags=!comment;]
```
Comment (KyleAure) [inline README.adoc]: @evelinec Combined the code snippets into this one area as instructed by issue here: https://github.com/OpenLiberty/guide-rest-client-java/issues/23

File: README.adoc [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2017 IBM Corporation and others.
+// Copyright (c) 2018 IBM Corporation and others.
```
Comment (evelinec) [inline README.adoc]: If you're updating an existing file, the copyright year should be `2017, 2018`, you can just change it to `2018`.

File: README.adoc [hunk 3]
```diff
@@ -163,76 +162,55 @@ http://json-b.net/users-guide.html[JSON Binding 1.0 Users Guide].
 
 == Consuming the REST resource
 
-Write a simple `Consumer` class that consumes the artist JSON using both JSON-B and JSON-P.
-
-`src/main/java/io/openliberty/guides/consumingrest/Consumer.java`:
+The `Artist` and `Album` POJOs are ready for deserialization. To consume the JSON response
+from your REST service create the `Consumer` class in the
```
Comment (evelinec) [inline README.adoc]: I think a comma is needed in "from your REST service, create the `Consumer` class in the"

File: README.adoc [hunk 4]
```diff
@@ -163,76 +162,55 @@ http://json-b.net/users-guide.html[JSON Binding 1.0 Users Guide].
...
+`src/main/java/io/openliberty/guides/consumingrest/Consumer.java` file.
```
Comment (evelinec) [inline README.adoc]: Put a ":" after file instead of "."

File: README.adoc [hunk 5]
```diff
@@ -163,76 +162,55 @@ http://json-b.net/users-guide.html[JSON Binding 1.0 Users Guide].
...
+use the custom constructors to create instances of `Artist` and `Album`.
 
 == Creating additional REST resources
 
 Now that you can consume a JSON resource you can put that data to use.
 Copy the `ArtistResource` class below and replace the `ArtistResource` class in
-the `start` directory.
-
-`src/main/java/io/openliberty/guides/consumingrest/service/ArtistResource.java`:
+the `src/main/java/io/openliberty/guides/consumingrest/service/ArtistResource.java` file.
```
Comment (evelinec) [inline README.adoc]: Use a ":" after file instead of "."

File: README.adoc [hunk 6]
```diff
@@ -251,28 +229,41 @@ in the JSON for a particular artist. The method returns -1 if this artist does n
...
 
 // =================================================================================================
 // Testing deserialization
 // =================================================================================================
 
 == Testing deserialization
 
-Create a test class, `src/test/java/it/io/openliberty/guides/consumingrest/ConsumingRestTest.java`.
+Create the `ConsumingRestTest` class in the
+`src/test/java/it/io/openliberty/guides/consumingrest/ConsumingRestTest.java` file.
```
Comment (evelinec) [inline README.adoc]: Use a ":" after file instead of "."

File: finish/src/main/java/io/openliberty/guides/consumingrest/Consumer.java [hunk 7]
```diff
@@ -1,6 +1,6 @@
 // tag::comment[]
 /*******************************************************************************
- * Copyright (c) 2017 IBM Corporation and others.
+ * Copyright (c) 2018 IBM Corporation and others.
```
Comment (evelinec) [inline finish/src/main/java/io/openliberty/guides/consumingrest/Consumer.java]: Same here for the copyright year.

**Comment by evelinec** [comment]
I'm okay with the change needed for issue #32.

**Comment by evelinec** [comment]
@Kubik42 can you review the other changes in this PR?

**Comment by KyleAure** [comment]
@ Kubik42 I actually did not write that part but it has been changed! 😆

**Comment by Kubik42** [comment]
I'm fixing my own mistakes then :P Looks ok to merge to me, thanks for all your work on this one @KyleAure

**Pull request merged**